raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -190,18 +190,18 @@ contract TokenNetwork is Utils {
         Channel storage channel = channels[channel_identifier];
         Participant storage participant_state = channel.participants[participant];
 
-        require(participant_state.deposit < total_deposit);
-
         // Calculate the actual amount of tokens that will be transferred
         added_deposit = total_deposit - participant_state.deposit;
 
         // Update the participant's channel deposit
         participant_state.deposit += added_deposit;
 
+        emit ChannelNewDeposit(channel_identifier, participant, participant_state.deposit);
+
         // Do the transfer
-        require(token.transferFrom(msg.sender, address(this), added_deposit));
+        assert(token.transferFrom(msg.sender, address(this), added_deposit));
```

Comment:
This should be a require(), not an assert. As per solidity's [docs](http://solidity.readthedocs.io/en/v0.4.23/control-structures.html?highlight=assert#error-handling-assert-require-revert-and-exceptions):

```
The require function should be used to ensure valid conditions, such as inputs, or contract state variables are met, or to validate return values from calls to external contracts.
```

Specifically note the `validate return values from calls to external contracts`